homeassistant/components/ads/sensor.py

```diff
@@ -32,6 +32,7 @@
                 ads.ADSTYPE_BYTE,
                 ads.ADSTYPE_DINT,
                 ads.ADSTYPE_UDINT,
+                ads.ADSTYPE_REAL,
```

Comment:
You are trying to change the `PLATFORM_SCHEMA`.
I think this is not allowed under [`ADR-0007`](https://github.com/home-assistant/architecture/blob/master/adr/0007-integration-config-yaml-structure.md) and the code needs to be refactored.

Note: I think [`ADR-0010`](https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md) also applies and a config-flow may need to be implemented.

Reply:
Hi @epenet,

thank you for the fast feedback. I'm quite new to home-assistant and just found that this missing feature is blocking me from usage in my home.

For clarification: To consist with [`ADR-0007`](https://github.com/home-assistant/architecture/blob/master/adr/0007-integration-config-yaml-structure.md) correctly this would mean the whole configuration for the `ads`-integration would need to be refactored, which would mean a huge breaking change for all existing installations?

Reply:
It would not be a breaking change.
See https://github.com/home-assistant/core/pull/87655 for how to implement a migration to config flow.
My main concern here is that I don't know if [ADR-0010](https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md) applies.

So either:
- you have to migrate to "configuration moved under the integration domain key"
- or you have to migrate to config-flow

Either way it's not a breaking change - it's a migration path with deprecation.